script.js

```diff
@@ -138,27 +137,24 @@ const editTask = (index) => {
 
 const saveTask = async (index, timeText, timeSum, timeDate) => {
   flagForEditing = -1;
-  text = timeText;
-  sum = timeSum;
-  date = timeDate;
-  if (!timeText.trim() || timeSum < 0 || !timeDate)
-    alert("пожалуйста, корректное значение");
-  else {
+  if (timeText.trim() || timeSum > 0 || timeDate) {
     const resp = await fetch(`http://localhost:7070/updateTasks`, {
       method: "PATCH",
       headers: {
         "Content-Type": "application/json;charset=utf-8",
         "Access-Control-Allow-Origin": "*",
       },
       body: JSON.stringify({
-        _id: allTasks[index],
-        text,
-        sum,
-        date,
+        _id: allTasks[index]._id,
+        text: timeText ? timeText : allTasks[index].text,
+        sum: timeSum ? timeSum : allTasks[index].sum,
+        date: timeDate ? timeDate : allTasks[index].date,
```

Comment:
it is ugly code. use another function for this, or use destructurisation and move it from the request to another place